### PR TITLE
Add Versionist & npm version scripts for automatic versioning

### PR DESCRIPTION
**Pull request by jhermsmeier**
With this, the following will run `versionist`, update the `CHANGELOG.md`, bump the `package.json` version, commit & tag:
```sh
npm run bump patch
```

Example changelog update diff:
```diff
+## v9.0.1 - 2017-03-01
+
+### Misc
+
+- **versionist:** Add changelog template
+- **versionist:** Add `getIncrementLevelFromCommit()`
+- **package:** Add "bump" script
+- **versionist:** Add versionist.conf.js
+- **package:** Add versionist & postversion hook
+- **editorconfig:** Disable trimming whitespace in Markdown (#89)
+- Update dependencies (#88)
+- Fix ENOSPC / EINVAL caused by chunk padding (#84)
+- **package:** Update dependencies (#82)
+- Update ImageStreamPosition unit tests (#81)
+
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,9 @@
   "scripts": {
     "test": "npm run lint && mocha --recursive tests/*.spec.js -R spec && node tests/e2e.js",
     "lint": "eslint lib tests bin example.js",
-    "readme": "jsdoc2md --template doc/README.hbs lib/index.js > README.md"
+    "readme": "jsdoc2md --template doc/README.hbs lib/index.js > README.md",
+    "bump": "npm version --tag-version-prefix \"\" -m \"v%s\"",
+    "version": "versionist"
```
Comment (jviotti) [inline package.json]: Can we name this task `changelog` for consistency purposes with other modules we have out there?

**Comment by jviotti** [comment]
Looks good to me. I'll invoke @hedss and @lekkas to also take a look, since they are more involved with Versionist than I am.

**Comment by hedss** [comment]
Looks good! But (there's always a but, isn't there?) I'd like @lekkas to comment, as we're diverging here from the 'usual process' documented in our commit/PR guidelines (only in-so-far as we're not using the same tags).

As far as the use of `VersionBot` is concerned, it's not an issue because the added `versionist.conf.js` will override the default process. However, this has been an ongoing conversation and Kostas does have a very strong opinion on this.

**Comment by jhermsmeier** [comment]
The tagging style changed at 3.0.0 (to non-prefixed with a "v"), not sure why – we can change it back to what the guidelines say.

File: package.json (hunk 1 above)
Comment (jhermsmeier) [inline package.json]: Well, then it wouldn't be run as the [`npm version`](https://docs.npmjs.com/cli/version) hook anymore (which does the tagging & commit etc). Could rename it to `changelog` and make the `"version": "npm run changelog"` though. What do you think?

**Comment by lurch** [comment]
Agreed that it makes sense (where possible / appropriate) to keep this as similar as possible to the "standard" resin.io versioning / versionist process.

**Comment by lekkas** [comment]
@jhermsmeier @hedss @jviotti the plan is to use the `versionbot` integration everywhere, but it's not there yet for `npm` modules. Until then, I'm fine with using versionist locally in projects and we can switch to `versionbot` later on.

The reason I'm pushing for a single `versionist.conf.js` managed by the versionbot integration is to keep the changelog and, most importantly, versioning conventions, consistent accross resin.io.

**Pull request merged**